node/src/Pos/Explorer/DB.hs

```diff
@@ -147,7 +151,8 @@ data ExplorerOp
 
     | PutPageBlocks !Page ![HeaderHash]
 
-    | PutEpochBlocks !Epoch ![HeaderHash]
+    | PutEpochBlocks !Epoch !Page ![HeaderHash]
```

Comment:
Yes, the database should be deleted and explorer restarted. 
The frontend is involved also, and there is no "middle ground" between migrations.

Reply:
> The frontend is involved also, and there is no "middle ground" between these migrations.

can you elaborate? (didn't get the point)

> Yes, the database should be deleted and explorer restarted.

I am ok for restarting the DB. Do we have to remove both nodes and Explorers DB or is removing Explorers DB sufficient ? Note that we won't be able to get away with removing DB all the time as it will increase sync time (thus increasing "downtime" )

At this moment I am not sure how Explorer will behave when its not fully synced:
 * how will main page behave ? (will it show older info or smth else)
 * how will address detail or transaction detail page behave ? (probably as address or tx can't be found)

@ksaric can you provide these details ^

Reply:
@akegalj I mean, the current DB format and the previous DB format are incompatible and the new format is bounded to the frontend directly - we have paging, we didn't have that before. 
This isn't just a simple DB migration, the web API has changed to reflect the changes.

We need to remove the explorer DB only. Until it's synced it will display the data that has at the moment - if it's syncing and currently has 200/2160 pages, it will display those 200 pages and all that they contain - address, transactions...

We could simply run this locally, connect explorer to _mainnet_, and then when it syncs, copy the DB to the "official" _mainnet_ server. It would need to sync only the blocks that have been created while we were transferring the DB from local to global.

Reply:
> This isn't just a simple DB migration, the web API has changed to reflect the changes.

I know web api has changed. Still, we will for sure modify Explorer DB scheme in the future. If its too complicated to migrate it now (because we still don't know how to properly migrate rocsdb) then we can remove the db and see how fast it will take for it to be up to date.

Can you verify this locally - it would be useful to know:
 1) how much time would it take to sync if we remove both node and explorer db
 2) how much time would it take to sync if we remove only explorer db (this would also ensure that removing only explorer db works as expected - as I never tried it myself)

can you do these tests? ^ 

@Pastafarianist would also like to hear your opinion about DB change, as you already did similar safe transition.

Reply:
1. I don't see any reason to try that and explorer DB is the node DB, since explorer is a node++.
2. Around the same speed as node syncing.

I don't why these results would be interesting and why we need to spend time on them, but I'll test it if it's necessary.
Again, we could find that out when we sync the DB before replacing it, if that's the direction we are going in.

Reply:
> I don't why these results would be interesting and why we need to spend time on them, but I'll test it if it's necessary.

If they differ a lot, then it might be beneficial to do trick with syncing and moving the db -> which complicates things for devops_

> Again, we could find that out when we sync the DB before replacing it, if that's the direction we are going in.

as said, this complicates life of devops and before doing so it would be useful to know is it really needed. Currently blockchain is not that big and we might sacrifice a minute of downtime in order not to change the deployment (if it takes a minute - which we would need to measure)

> I don't see any reason to try that and explorer DB is the node DB, since explorer is a node++.

this implies that explorer db is the same one as nodes db ? (have to check it in code)

Reply:
> this implies that explorer db is the same one as nodes db ? (have to check it in code)

AFAIK explorer DB should be node DB + explorer specific part(s).

Reply:
@ksaric can't we extend `prepareExplorerDB` in a way that adds new `Page` to the DB ? We can iterate over epoch and calculate this `Page` and fix the DB ?

Reply:
@akegalj And save it where? Notice that the prefix/key is the same, just a `Page` was added - https://github.com/input-output-hk/cardano-sl/pull/1695/files#diff-9ea213700be61c330d86a770f61d6d03L261
I'm not saying it's not possible, but would be quite difficult - for example, do we fit all the current epoch blocks in memory, if we go down that route? 
I _could_ change the key/prefix, start iterating using conduit and place that in the database. 
But then we would have a check that checks that once, not a DB migration. 
It's likely to take some time, and the end result? A piece of code that's used _once_.